### DISCUSSION TITLE
IPv6: DHCPv6 vs SLAAC

### DISCUSSION THREAD

**Discussion by fakemanhk** [fakemanhk/openwrt-jp-ipoe#2] [Q&A]
There are some issues with Windows 10, Chrome OS and Android clients, only link-local IPv6 address is assigned to those devices.

**Comment by fakemanhk** [comment]
When I tested this with a Windows 10 laptop, to my surprise that it was not able to get GUA from DHCPv6, this seems to be a known issue and I followed [this page](https://ipv6.web.cern.ch/content/ms-windows-client-doesnt-get-ipv6-address-dhcpv6) to configure my Windows and IPv6 address can be assigned properly with DHCPv6.

The setup I learned from local Japanese sites has no issue with Linux/iOS devices as well, however I noticed that Chrome OS and Android devices are not able to get proper IPv6 address, due to the nonexistence support of DHCPv6 [Decided by Google](https://issuetracker.google.com/issues/36949085), so there is no fix/workaround), if you have any Chrome OS/Android device, you will need to enable _**SLAAC**_ on LAN interface.

## Steps to configure LAN interface with SLAAC:

![](https://user-images.githubusercontent.com/21307353/213613705-6c40902c-3d29-4186-80a9-f11a73d27ac6.png)

This time I make LAN side RA service running in _**Server Mode**_, and disable DHCPv6, it will then show you the **IPv6 RA Setting** tab.

![](https://user-images.githubusercontent.com/21307353/213614018-f3a14c79-25a8-4620-8df9-59d1a82ddc88.png)

Check the box _**Enable SLAAC**_, and make sure _**no RA Flags**_ being selected.

Reconnect Android / Chrome OS devices, GUA with same prefix as WAN6 interface should be assigned without problem.

**Reply by Soromeister** [reply]
Disabling `DHCPv6-Service` will cause all computers with Windows and macOS to not be listed under the `Active DHCPv6 Leases` section in LuCi.
Disabling `RA-Flags` will cause all computers with macOS to not be listed under the `Active DHCPv6 Leases` section in LuCi.
They do get IPv6 addresses though.

**Comment by fakemanhk** [comment]
> Disabling `DHCPv6-Service` will cause all computers with Windows and macOS to not be listed under the `Active DHCPv6 Leases` section in LuCi. Disabling `RA-Flags` will cause all computers with macOS to not be listed under the `Active DHCPv6 Leases` section in LuCi. They do get IPv6 addresses though.

Your observation is correct, so it depends on how you want your home network look likes.
As mentioned, ChromeOS & Android are not supporting DHCPv6 (stateful), some OSes by default also have certain problem when dealing with DHCPv6, while SLAAC is the standard way which supported by all OS. So here is the problem: When you have Apple devices + Android/ChromeOS at home, you still cannot see the later one from DHCPv6 leases, apparently it will confuse people that some devices "vanished".

I can understand the point that users might want a better control on IP distribution for management, however due to the default randomized MAC address nature of modern devices, the privacy extension, you won't get much from DHCPv6, unless you have something like domain management inside company that can control them.

**Comment by zzl221000** [comment]
Hey bro
I currently have a switch that uses SLAAC to assign IPv6 addresses. The switch limits the number of ND (Neighbor Discovery) entries to a maximum of 65,535. However, I have 100,000 devices. I'm considering switching to OpenWRT. Does OpenWRT have any limitations on the number of ND entries?
